lib/utils/repoGitDiff.js

```diff
@@ -3,22 +3,31 @@ const childProcess = require('child_process')
 const cpUtils = require('./childProcessUtils')
 const gc = require('./gitConstants')
 const os = require('os')
-const path = require('path')
+const fs = require('fs')
+const path = require('path');
+const ignore = require('ignore')
 
 const fullDiffParams = ['diff', '--name-status', '--no-renames']
+const ig = ignore()
 
 module.exports = (config, metadata) => {
   const { stdout: diff } = childProcess.spawnSync(
     'git',
     [...fullDiffParams, config.from, config.to],
-    { cwd: config.repo, encoding: gc.UTF8_ENCODING }
+    { cwd: config.repo, encoding: gc.UTF8_ENCODING}
   )
+
+  if (config.ignore && fs.existsSync(config.ignore)) {
+    ig.add(fs.readFileSync(config.ignore).toString())
+  }
+
   return treatResult(cpUtils.treatDataFromSpawn(diff), metadata)
 }
 
 const treatResult = (repoDiffResult, metadata) =>
   repoDiffResult
     .split(os.EOL)
+    .filter(line => (line ? !ig.ignores(line) : false))
```

Comment:
`
.filter(line => !ig.ignores(line))
`
Could work ?

Reply:
Unfortunately line was sometimes null/undefined and ignores would throw an error.

Reply:
So we could rewrite it like this : 
`
.filter(line => line && !ig.ignores(line))
`
?

Reply:
ahh yes of course. that should work.  I'll update.